### PR TITLE
[tapocontrol] included P115 smart-socket

### PR DESCRIPTION
**Pull request by wildcs**
This PR now added the Tapo P115 smart-socket to tapocontrol.
Also fixed thing configuration warning occured since OH3.3

### CODE REVIEW

**Comment by lolodomo** [comment]
> Also fixed thing configuration warning occured since OH3.3

The configuration parameter is defined as an integer, there is no real sense to consider in the binding it could be a decimal.
Maybe the root cause leading to a decimal value in your setup should rather be determined and fixed.

If not, this would mean we have to fix all bindings to habdle something that should not happen.

**Comment by lolodomo** [comment]
Could exclude the new parameter handling and put it in a separate PR so that I can at least merge the PR which is about adding P115 smart socket ?

File: bundles/org.openhab.binding.tapocontrol/src/main/java/org/openhab/binding/tapocontrol/internal/structures/TapoBridgeConfiguration.java [hunk 1]
```diff
@@ -68,7 +68,7 @@ public void loadSettings() {
             username = config.get(CONFIG_EMAIL).toString();
             password = config.get(CONFIG_PASS).toString();
             cloudDiscoveryEnabled = Boolean.parseBoolean(config.get(CONFIG_DISCOVERY_CLOUD).toString());
-            discoveryIntervalM = Integer.valueOf(config.get(CONFIG_DISCOVERY_INTERVAL).toString());
+            discoveryIntervalM = (int) Double.parseDouble(config.get(CONFIG_DISCOVERY_INTERVAL).toString());
```
Comment (jlaur) [inline bundles/org.openhab.binding.tapocontrol/src/main/java/org/openhab/binding/tapocontrol/internal/structures/TapoBridgeConfiguration.java:71]: This work-around should not be needed in 3.4 as fixed by openhab/openhab-core#3024. See also #13112.

File: bundles/org.openhab.binding.tapocontrol/src/main/java/org/openhab/binding/tapocontrol/internal/structures/TapoDeviceConfiguration.java [hunk 2]
```diff
@@ -55,7 +55,7 @@ public void loadSettings() {
         try {
             Configuration config = this.device.getConfiguration();
             this.ipAddress = config.get(CONFIG_DEVICE_IP).toString();
-            this.pollingInterval = Integer.valueOf(config.get(CONFIG_UPDATE_INTERVAL).toString());
+            this.pollingInterval = (int) Double.parseDouble(config.get(CONFIG_UPDATE_INTERVAL).toString());
```
Comment (jlaur) [inline bundles/org.openhab.binding.tapocontrol/src/main/java/org/openhab/binding/tapocontrol/internal/structures/TapoDeviceConfiguration.java:58]: This work-around should not be needed in 3.4 as fixed by openhab/openhab-core#3024. See also #13112.

File: bundles/org.openhab.binding.tapocontrol/README.md [hunk 3]
```diff
@@ -12,7 +12,7 @@ The following Tapo-Devices are supported
 * Wi-Fi signal (SignalStrength)
 * On-Time (Time in seconds device is switched on)
 
-### P110 EnergyMonitoring SmartPlug (WiFi)
+### P110/P115 EnergyMonitoring SmartPlug (WiFi)
```
Comment (jlaur) [inline bundles/org.openhab.binding.tapocontrol/README.md:15]:
```suggestion
### P110/P115 EnergyMonitoring SmartPlug (Wi-Fi)
```

**Review by jlaur — CHANGES_REQUESTED** [review]
Thanks! I have added a few minor comments.

**Comment by jlaur** [comment]
@lolodomo - sorry, I missed your comments when checking the PR. At least our comments are consistent. :)

**Review by jlaur — APPROVED** [review]
LGTM

**Comment by wildcs** [comment]
removed the parameter handling workaround.
But without that i get still an error:
`NumberFormatException@203 "java.lang.NumberFormatException: For input string: "20.0"`
So the setting is useless?

**Comment by jlaur** [comment]
@wildcs - I assume you are running the binding in openHAB 3.3? The mentioned bugfix is included in 3.4. You could try to run a snapshot version to verify this. Probably the first 3.4 milestone release is not far away.

**Comment by wildcs** [comment]
@jlaur - no i'm using OH3.4 Build #3034 on my coding pc

**Comment by jlaur** [comment]
> no i'm using OH3.4 Build 3034 on my coding pc and there i can reproduce and debug the error.

Sorry, I missed that openhab/openhab-core#3024 is not merged yet. So we expect to have a generic fix for 3.4 so that we don't have to fix all bindings, but the PR needs review with the possibility of more work to be done.

**Comment by jlaur** [comment]
> > no i'm using OH3.4 Build 3034 on my coding pc and there i can reproduce and debug the error.
> 
> Sorry, I missed that [openhab/openhab-core#3024](https://github.com/openhab/openhab-core/pull/3024) is not merged yet. So we expect to have a generic fix for 3.4 so that we don't have to fix all bindings, but the PR needs review with the possibility of more work to be done.

openhab/openhab-core#3024 was recently merged, so this should be fixed now. Perhaps you could verify with latest snapshot - otherwise it will be included in next milestone.

**Comment by wildcs** [comment]
Hi,
testet with Snapshot #3103. Issue is still there when reading "discoveryIntervalM" :
image
Object x is for debugging only

**Comment by jlaur** [comment]
> Hi,
> testet with Snapshot #3103. Issue is still there when reading "discoveryIntervalM"

Did you test with latest snapshot of core, or only binding?

**Comment by wildcs** [comment]
Downloaded latest snapshot from oh-homepage and put it into debugging enviroment
image

**Pull request merged**